src/contracts/account/ReplayProtection.sol

```diff
@@ -67,42 +82,58 @@ contract ReplayProtection {
      * MultiNonce replay protection.
      * Explained: https://github.com/PISAresearch/metamask-comp#multinonce
      * Allows a user to send N queues of transactions, but transactions in each queue are accepted in order.
-     * If nonce1==0, then it is the same as replace-by-version (e.g. increment nonce each time).
+     * If queue==0, then it is a single queue (e.g. NONCE replay protection)
      * @param _replayProtection Contains a single nonce
      */
     function nonce(address _signer, bytes memory _replayProtection) internal returns(bool) {
-        uint256 nonce1;
-        uint256 nonce2;
+        uint256 queue;
+        uint256 queueNonce;
 
-        (nonce1, nonce2) = abi.decode(_replayProtection, (uint256, uint256));
-        bytes32 index = keccak256(abi.encode(_signer, nonce1));
+        (queue, queueNonce) = abi.decode(_replayProtection, (uint256, uint256));
+        bytes32 index = queueIndex(_signer, queue, getMultiNonceAddress());
         uint256 storedNonce = nonceStore[index];
 
         // Increment stored nonce by one...
-        if(nonce2 == storedNonce) {
+        if(queueNonce == storedNonce) {
             nonceStore[index] = storedNonce + 1;
             return true;
         }
 
         return false;
     }
-
     /**
      * Bitflip Replay Protection
      * Explained: https://github.com/PISAresearch/metamask-comp#bitflip
-     * Allows a user to flip a bit in nonce2 as replay protection. Every nonce supports 256 bit flips.
+     * Signer flips a bit for every new transaction. Each queue supports 256 bit flips.
      */
     function bitflip(address _signer, bytes memory _replayProtection) internal returns(bool) {
-        (uint256 nonce1, uint256 bitsToFlip) = abi.decode(_replayProtection, (uint256, uint256));
+        (uint256 queue, uint256 bitsToFlip) = abi.decode(_replayProtection, (uint256, uint256));
+
+        require(bitsToFlip > 0, "It must flip one bit!");
 
-        // It is unlikely that anyone will need to send 6174 concurrent transactions per block,
-        // plus 6174 is a cool af number.
-        require(nonce1 >= 6174, "Nonce1 must be at least 6174 to separate multinonce and bitflip");
+        // n & (n-1) == 0, i.e. is it a power of two?
+        // example: 4 = 100, 3 = 011. 4 & = 000.
+        require(bitsToFlip & bitsToFlip-1 == 0, "Only a single bit can be flipped");
```

Comment:
I was about to suggest that you should have a test for "more bits flipped should fail", but you win this time. :P

Reply:
done